tests/compile/test_fusion_attn.py

```diff
@@ -296,28 +314,49 @@ def forward(self, q: torch.Tensor, k: torch.Tensor, v: torch.Tensor):
                                      out_dtype=attn_output.dtype)
 
 
-@pytest.mark.parametrize("num_qo_heads, num_kv_heads", [(64, 8), (40, 8)])
+CUDA_MODELS = [("nvidia/Llama-4-Scout-17B-16E-Instruct-FP8",
+                TestAttentionFp8StaticQuantPatternModel),
+               ("nvidia/Llama-4-Scout-17B-16E-Instruct-FP4",
+                TestAttentionNvfp4QuantPatternModel)]
+ROCM_MODELS = [("amd/Llama-3.1-8B-Instruct-FP8-KV",
+                TestAttentionFp8StaticQuantPatternModel)]
+
+CUDA_HEADS = [(64, 8), (40, 8)]
+ROCM_HEADS = [(32, 8), (40, 8)]
+
+
+@pytest.mark.parametrize(
+    "num_qo_heads, num_kv_heads",
+    CUDA_HEADS if current_platform.is_cuda() else ROCM_HEADS)
 @pytest.mark.parametrize("head_size", [128])
-@pytest.mark.parametrize("batch_size", [7, 256, 533])
+@pytest.mark.parametrize("batch_size",
+                         [7, 256, 533] if current_platform.is_cuda() else [8])
 @pytest.mark.parametrize("dtype", [torch.bfloat16])
-@pytest.mark.parametrize("model_name, model_class",
-                         [("nvidia/Llama-4-Scout-17B-16E-Instruct-FP8",
-                           TestAttentionFp8StaticQuantPatternModel),
-                          ("nvidia/Llama-4-Scout-17B-16E-Instruct-FP4",
-                           TestAttentionNvfp4QuantPatternModel)])
-@pytest.mark.parametrize("backend", [_Backend.FLASHINFER])
-@pytest.mark.skipif(not current_platform.is_cuda(), reason="Only test CUDA")
+@pytest.mark.parametrize(
+    "model_name, model_class",
+    CUDA_MODELS if current_platform.is_cuda() else ROCM_MODELS)
+@pytest.mark.parametrize("backend", [_Backend.FLASHINFER] if
+                         current_platform.is_cuda() else [_Backend.ROCM_FLASH])
+@pytest.mark.parametrize(
+    "split_attention",
+    [False, True] if current_platform.is_rocm() else [False])
+@pytest.mark.skipif(not current_platform.is_cuda_alike(),
+                    reason="Only test ROCm or CUDA")
 @pytest.mark.skipif(not current_platform.supports_fp8(), reason="Need FP8")
-@pytest.mark.skipif(not current_platform.is_device_capability((10, 0)),
-                    reason="Only test on SM100(Blackwell)")
+@pytest.mark.skipif(current_platform.is_cuda()
+                    and not current_platform.is_device_capability((10, 0)),
+                    reason="On CUDA only test on SM100(Blackwell)")
```

Comment:
Also need to make sure this is skipped on non-cuda-alike platforms:
```suggestion
@pytest.mark.skipif(not current_platform.is_cuda_like())
@pytest.mark.skipif(current_platform.is_cuda()
                    and not current_platform.is_device_capability((10, 0)),
                    reason="On CUDA only test on SM100(Blackwell)")
```